cmplibrary/src/main/java/com/sourcepoint/gdpr_cmplibrary/StoreClient.java

```diff
@@ -48,36 +55,34 @@
 
     public static final String DEFAULT_AUTH_ID = null;
 
+    public static final String TC_KEYS_DELIMITER = ";";
 
     StoreClient(SharedPreferences pref){
         this.editor = pref.edit();
         this.pref = pref;
     }
 
-    public void setConsentSubjectToGDPr(Boolean consentSubjectToGDPR){
-        editor.putString(IAB_CONSENT_SUBJECT_TO_GDPR, consentSubjectToGDPR != null ? (consentSubjectToGDPR ? "1" : "0") : null);
-        editor.commit();
-    }
-
-    public void setIabConsentCmpPresent(Boolean iabConsentCmpPresent){
-        editor.putBoolean(IAB_CONSENT_CMP_PRESENT, iabConsentCmpPresent);
-        editor.commit();
-    }
-
-    public void setIabConsentConsentString(String consentConsentString){
-        editor.putString(IAB_CONSENT_CONSENT_STRING, consentConsentString);
-        editor.putString(EU_CONSENT__KEY, consentConsentString);
-        editor.commit();
-    }
-
-    public void setIabConsentParsedPurposeConsents(String consentParsedPurposeConsents){
-        editor.putString(IAB_CONSENT_PARSED_PURPOSE_CONSENTS, consentParsedPurposeConsents);
+    public void setTCData(HashMap<String, String> data){
+        for(String s : data.keySet()){
+            editor.putString(s, data.get(s));
+        }
+        editor.putString(TC_KEYS_KEY, dataSetToStr(data.keySet()));
```

Comment:
Why do we need the `TC_KEYS_KEY`?
So we can remove all IAB data later on?

Reply:
Yes, and also in order to get the TCData back from the local storage.